dffml/source/json.py

```diff
@@ -20,13 +20,22 @@ class JSONSource(FileSource, MemorySource):
     async def load_fd(self, fd):
         repos = json.load(fd)
         self.mem = {
-            src_url: Repo(src_url, data=data)
-            for src_url, data in repos.items()
+            self.config.label: [
+                {
+                    src_url: Repo(src_url, data=data)
+                    for src_url, data in repos.items()
+                }
+            ]
         }
         LOGGER.debug("%r loaded %d records", self, len(self.mem))
 
     async def dump_fd(self, fd):
         json.dump(
-            {repo.src_url: repo.dict() for repo in self.mem.values()}, fd
+            {
+                self.config.label: [
+                    {repo.src_url: repo.dict() for repo in self.mem.values()}
+                ]
+            },
+            fd,
```

Comment:
You need to load the file with `json.load` then update the label within the loaded JSON, then dump the modified version back into the file. My guess would be that this is how it should be done:

```python
repos = json.load(fd)
repos[self.config.label] = {repo.src_url: repo.dict() for repo in self.mem.values()}
fd.seek(0)
json.dump(repos, fd)
```